landoapi/models/landing.py

```diff
@@ -128,3 +132,38 @@ class RevisionNotFoundException(Exception):
     def __init__(self, revision_id):
         super().__init__()
         self.revision_id = revision_id
+
+
+def _upload_patch_to_s3(patch, revision_id, diff_id):
+    """Save patch in S3 bucket.
+
+    Creates a temporary file and uploads it to an S3 bucket.
+    Requires PATCH_BUCKET_NAME to be provided as an environment variable.
+
+    Args:
+        patch: Text to be saved
+        revision_id: String ID of the revision (ex. 'D123')
+        diff_id: The integer ID of the raw diff
+
+    Returns
+        String representing the patch's URL in S3
+        (ex. 's3://{bucket_name}/D123_1.patch')
+    """
+    s3 = boto3.resource(
+        's3',
+        # access and secret should be only provided for development
+        aws_access_key_id=os.getenv('AWS_ACCESS_KEY', None),
+        aws_secret_access_key=os.getenv('AWS_SECRET_KEY', None)
+    )
+    bucket = os.getenv('PATCH_BUCKET_NAME')
+    patch_name = '{revision_id}_{diff_id}.patch'.format(
+        revision_id=revision_id, diff_id=diff_id
+    )
+    patch_url = 's3://{bucket}/{patch_name}'.format(
+        bucket=bucket, patch_name=patch_name
+    )
+    with tempfile.TemporaryFile() as patchfile:
+        patchfile.write(patch.encode('utf-8'))
+        patchfile.seek(0)
+        s3.meta.client.upload_fileobj(patchfile, bucket, patch_name)
```

Comment:
Uploading this file appears to be a significant action - does it need an INFO level log message?